File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode83.java
class ListNode {
    int data;
    ListNode next;

    ListNode(int data) {
        this.data = data;
        this.next = null;
    }

    ListNode(int data, ListNode next) {
        this.data = data;
        this.next = next;
    }
}

public class Leetcode83 {
    public static void main(String[] args) {
        ListNode head = new ListNode(1);
        addLast(2, head);
        addLast(2, head);
        addLast(4, head);
        addLast(5, head);
        addLast(6, head);
        printList(head);
        System.out.println("After removing");
        removeDuplicate(head);
        printList(head);
    }

    private static void addLast(int data, ListNode head) {
        ListNode currNode = head;
        while (currNode.next != null) {
            currNode = currNode.next;
        }
        currNode.next = new ListNode(data);
    }

    private static void printList(ListNode head) {
        ListNode currNode = head;
        while (currNode != null) {
            System.out.print(currNode.data + " --> ");
            currNode = currNode.next;
        }
        System.out.println("NULL");
    }

    private static ListNode removeDuplicate(ListNode head) {
        ListNode currNode = head;
        while (currNode != null && currNode.next != null) {
            if(currNode.data == currNode.next.data) {
                currNode.next = currNode.next.next;
            }else{
                currNode = currNode.next;
            }
        }
        return head;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode83.ts
class ListNode {
    val: number
    next: ListNode | null
    constructor(val?: number, next?: ListNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.next = (next === undefined ? null : next)
    }
}

function deleteDuplicates(head: ListNode | null): ListNode | null {
    let currNode: ListNode | null = head;
    while (currNode != null && currNode.next != null) {
        if (currNode.val == currNode.next.val) {
            currNode.next = currNode.next.next;
        } else {
            currNode = currNode.next;
        }
    }
    return head;
};

function addLast(head: ListNode, data: number): void {
    let currNode: ListNode | null = head;
    while (currNode?.next != null) {
        currNode = currNode.next;
    }
    currNode.next = new ListNode(data);
}

function printList(head: ListNode): void {
    let currNode: ListNode | null = head;
    let result: string = "";
    while (currNode != null) {
        result += `${currNode.val} --> `;
        currNode = currNode.next;
    }
    result += "Null";
    console.log(result);
}

let head: ListNode = new ListNode(1);
addLast(head, 2);
addLast(head, 2);
addLast(head, 3);
addLast(head, 3);
addLast(head, 4);
addLast(head, 4);
addLast(head, 5);
addLast(head, 5);
addLast(head, 6);
addLast(head, 6);
printList(head);
console.log("After Removing: ");
deleteDuplicates(head);
printList(head);

export { };